android/build.sh

```diff
@@ -45,41 +105,108 @@ clean ( ) {
   fi
 }
 
-echo Build KMEA and KMAPro:
+function _build_engine() {
+  cd "$KEYMAN_ROOT/android/KMEA"
+  ./build.sh $KMEA_FLAGS
+
+  if [ $? -ne 0 ]; then
+    die "ERROR: KMEA/build.sh failed"
+  fi
+}
+
+function _build_app() {
+  cd "$KEYMAN_ROOT/android/KMAPro"
+  ./build.sh $KMAPRO_FLAGS
+
+  if [ $? -ne 0 ]; then
+    die "ERROR: KMAPro/build.sh failed"
+  fi
+}
+
+function _build_samples() {
+  cd "$KEYMAN_ROOT/android/Samples/KMSample1"
+  ./build.sh $SAMPLE_FLAGS
+
+  if [ $? -ne 0 ]; then
+    die "ERROR: KMSample1/build.sh failed"
+  fi
+
+  cd "$KEYMAN_ROOT/android/Samples/KMSample2"
+  ./build.sh SAMPLE_FLAGS
+
+  if [ $? -ne 0 ]; then
+    die "ERROR: KMSample2/build.sh failed"
+  fi
+
+  cd "$KEYMAN_ROOT/android/Tests/KeyboardHarness"
+  ./build.sh $SAMPLE_FLAGS
+
+  if [ $? -ne 0 ]; then
+    die "ERROR: KeyboardHarness/build.sh failed"
+  fi
+}
+
+function _build_fv() {
+  pushd "$KEYMAN_ROOT/oem/firstvoices/android"
+  ./build.sh $FV_FLAGS
+
+  if [ $? -ne 0 ]; then
+    die "ERROR: oem/firstvoices/android/build.sh failed"
+  fi
+}
 
 # Check about cleaning artifact paths
-if [[ "$1" == "-clean" ]] ; then
-  clean
-  exit
+if builder_start_action clean; then
+  _clean
+  builder_finish_action success clean
 fi
 
 # Building Keyman Engine for Android
-
-cd "$KEYMAN_ROOT/android/KMEA"
-./build.sh "$@"
-
-if [ $? -ne 0 ]; then
-    die "ERROR: KMEA/build.sh failed"
+if builder_start_action build:engine; then
+  _build_engine
+  builder_finish_action success build:engine
 fi
 
 # Building Keyman for Android
+if builder_start_action build:app; then
+  _build_app
+  builder_finish_action success build:app
+fi
 
-cd "$KEYMAN_ROOT/android/KMAPro"
-./build.sh "$@"
+# Default build action
+if builder_start_action build; then
+  _build_engine
+  _build_app
+  builder_finish_action success build
```

Comment:
I think I saw one of the CI Android Test-14.0 builds (`./build.sh build`)attempt to start building Samples/KMSample1.
Does the the default `build` action always attempt every target?

Reply:
> Does the the default `build` action always attempt every target?

Yes, by design. You should specify `build:<target>` to only build a specific target.

Reply:
Good to hear. Will review with you next sprint. Also, am doing some work on builder scripts next week to support some of @jahorton's recent work in #7392, so that may have an impact on you too.